framework/src/consumption/lib/opensearch/opensearch.ts

```diff
@@ -294,31 +296,66 @@ export class OpenSearchCluster extends TrackedConstruct {
       },
       vpc: this.vpc,
       subnets: subnets,
+      removalPolicy: this.removalPolicy,
     });
 
     this.domain = domain;
 
     const samlAdminGroupId = props.samlMasterBackendRole;
 
-    this.addRoleMapping('AllAccessOsRole', 'all_access', samlAdminGroupId);
-    this.addRoleMapping('SecurityManagerOsRole', 'security_manager', samlAdminGroupId);
+    this.addRoleMapping('AllAccessRoles', 'all_access', [samlAdminGroupId, this.masterRole.roleArn], true);
+    this.addRoleMapping('SecurityManagerRoles', 'security_manager', [samlAdminGroupId, this.masterRole.roleArn], true);
+
+    //enable SAML authentication after adding lambda permissions to execute API calls later.
+    const updateDomain = new AwsCustomResource(this, 'EnableInternalUserDatabaseCR', {
+      onCreate: {
+        service: 'OpenSearch',
+        action: 'updateDomainConfig',
+        parameters: {
+          DomainName: this.domain.domainName,
+          AdvancedSecurityOptions: {
+            SAMLOptions: {
+              Enabled: true,
+              Idp: {
+                EntityId: samlMetaData.idpEntityId,
+                MetadataContent: samlMetaData.idpMetadataContent,
+              },
+              MasterBackendRole: samlMetaData.masterBackendRole,
+              RolesKey: samlMetaData.rolesKey,
+              SessionTimeoutMinutes: samlMetaData.sessionTimeoutMinutes,
+              SubjectKey: samlMetaData.subjectKey,
+            },
+          },
+        },
+        physicalResourceId: PhysicalResourceId.of('idpEntityId'),
+        outputPaths: ['DomainConfig.AdvancedSecurityOptions.SAMLOptions'],
+      },
+      role: this.masterRole,
+      removalPolicy: this.removalPolicy,
+    });
+    updateDomain.node.addDependency(this.domain);
+    if (this.prevCr) updateDomain.node.addDependency(this.prevCr);
+    this.prevCr = updateDomain;
   }
 
   /**
    * Calls OpenSearch API using custom resource.
    * @param id The CDK resource ID
    * @param apiPath  OpenSearch API path
    * @param body  OpenSearch API request body
+   * @param method Opensearch API method, @default PUT
    */
 
-  public callOpenSearchApi(id: string, apiPath: string, body: any) {
+  public callOpenSearchApi(id: string, apiPath: string, body: any, method?: string) {
     const cr = new CustomResource(this, id, {
       serviceToken: this.apiProvider.serviceToken,
       resourceType: 'Custom::OpenSearchAPI',
       properties: {
         path: apiPath,
         body,
+        method: method ?? 'PUT',
       },
+      removalPolicy: this.removalPolicy,
```

Comment:
We need to be careful here, can you add this as a parameter? This is a hidden behavior for the user, we need to give them the ability to either retain or delete on a per resource level when it comes to these calls.